### ISSUE TITLE
Wonky conditional in astropy.io.ascii.core

### ISSUE THREAD

**Issue by josePhoenix** [astropy/astropy#5336]
Chanced across some very unusual code in astropy.io.ascii.core. I don't think it's responsible for my colleague's issue, but I believe it should be looked at as it almost certainly does not work as intended.

In #910, this code was added to `core.py` starting at [line 745](https://github.com/astropy/astropy/blob/master/astropy/io/ascii/core.py#L745-L749):

```python
            # when we write tables the columns may be astropy.table.Columns
            # which don't carry a fill_values by default
            for col in cols:
                if ~hasattr(col, 'fill_values'):
                    col.fill_values = {}
```

If you were curious like me, this is *not* a TypeError. In fact...

```python
In [1]: ~True
Out[1]: -2

In [2]: ~False
Out[2]: -1

In [3]: if ~True: print("ok")
ok

In [4]: if ~False: print("ok")
ok
```

So, that branch of code will always run, whether or not the column has a `fill_values` attribute.

**Comment by MSeifert04** [comment]
Thank you for the report. This seems like an obvious mistake. Good catch :+1: 

I just checked and replacing it with `not hasattr( ... )` doesn't break the test suite (https://travis-ci.org/MSeifert04/astropy/builds/160318282). However I'm not sure how to test this path and if this could produce a Bug. Do you want to open a PR to fix that?

> I don't think it's responsible for my colleague's issue.

Could you clarify which issue? Is there already a filed issue?

**Comment by josePhoenix** [comment]
I don't currently have time to open that PR, so please go ahead if you have a source checkout you were testing with. Otherwise I can return to this another time. I believe it would be a bug in that fill values specified on Columns would always get clobbered with an empty dict.

There is another issue where masked tables with a masked value in the first row first column don't survive a round trip through ECSV format. I haven't had time to identify a minimal test case, and my colleague is working around it for now, so I haven't opened an issue report yet.

**Comment by eteq** [comment]
@MSeifert04 - I know this is already closed via #5338, but any chance you can write a quick regression test for this (or @josePhoenix as the original reporter of the issue)?  Clearly it's bad that the test suite never caught this!

**Comment by MSeifert04** [comment]
@eteq I'll submit a regression test later.
